### PR TITLE
Import and extend superfluous parens examples

### LINKED ISSUE #6226: Adjust unnecessarily precise description of superfluous-parens message
### Bug description

My initial code was:
```
if (i - 0) in exclude:
    continue
````
Which trigger superfluous-parens. I think the parentheses are technically unnecessary but it doesn't seem match with the intended target of this warning:
```
if (x == y):
    pass
```
I adjusted my code to:
```
foo = True
if foo and (i - 0) in exclude:
    continue
``` 
Which does not trigger superfluous-parens.
This makes me think that the warning is reacting to the proximity of the `if` and the `(` not the context in which they are being used.

### Configuration

_No response_

### Command used

```shell
python3 -m pylint --exit-zero --rcfile=pylint-config.txt scripts/tamper.py
```

### Pylint output

```shell
scripts/tamper.py:152:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
```

### Expected behavior

No superfluous-parens warning.

### Pylint version

```shell
pylint 2.12.2
```

### OS / Environment

macOS Monterey 12.2.1

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by mpernigo**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description
Copy examples from [pylint-errors - C0325](https://vald-phoenix.github.io/pylint-errors/plerr/errors/format/C0325.html) and extend them with the case raised on #6226.

Refs #5953

Closes #6226

### CODE REVIEW

File: doc/data/messages/s/superfluous-parens/bad.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+x = input()
+y = input()
+if (x == y):
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/s/superfluous-parens/bad.py:3]:
```suggestion
if (x == y):  # [superfluous-parens]
```

File: doc/data/messages/s/superfluous-parens/bad.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    pass
+
+i = 0
+exclude = []
+if (i - 0) in exclude:
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/s/superfluous-parens/bad.py:8]:
```suggestion
if (i - 0) in exclude:  # [superfluous-parens]
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you 👍

**Comment by Pierre-Sassoulas** [comment]
Congratulation on becoming a pylint contributor 😄 !

**Comment by mpernigo** [comment]
> Congratulation on becoming a pylint contributor 😄 !

Thank you 🎉

**Pull request merged**